tests/std/tests/P1020R1_smart_pointer_for_overwrite/test.cpp

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <cstdlib>
+#include <memory>
+#include <new>
+#include <stdexcept>
+#include <type_traits>
+#include <vector>
+
+using namespace std;
+
+size_t allocationCount = 0;
+
+struct ReportAddress;
+vector<ReportAddress*> ascendingAddressBuffer;
+vector<ReportAddress*> descendingAddressBuffer;
+
+#pragma warning(disable : 28251) // Inconsistent annotation for 'new'
+
+// According to N4849, the default behavior of operator new[](size) is to return
+// operator new(size), so only the latter needs to be replaced.
+void* operator new(size_t size) {
+    void* const p = ::operator new(size, nothrow);
+
+    if (p) {
+        return p;
+    } else {
+        throw bad_alloc();
+    }
+}
+
+void* operator new(size_t size, const nothrow_t&) noexcept {
+    void* const result = malloc(size == 0 ? 1 : size);
+    ++allocationCount;
+    if (result) {
+        memset(result, 0xEE, size);
+    }
+
+    return result;
+}
+
+void* operator new(size_t size, align_val_t align) {
+    void* const p = ::operator new(size, align, nothrow);
+
+    if (p) {
+        return p;
+    } else {
+        throw bad_alloc();
+    }
+}
+
+void* operator new(size_t size, align_val_t align, const nothrow_t&) noexcept {
+    void* const result = ::_aligned_malloc(size, static_cast<size_t>(align));
+    ++allocationCount;
+    if (result) {
+        memset(result, 0xEE, size);
+    }
+
+    return result;
+}
+
+template <typename T, typename = void>
+struct unique_is_for_overwritable : false_type {};
+
+template <typename T>
+struct unique_is_for_overwritable<T, void_t<decltype(make_unique_for_overwrite<T>())>> : true_type {};
+
+template <typename T>
+constexpr bool unique_is_for_overwritable_v = unique_is_for_overwritable<T>::value;
+
+struct DefaultInitializableInt {
+    int value;
+    DefaultInitializableInt() : value(106) {}

```

Comment:
Could you turn the magic number into a const, with a readable name? It doesn't change the functionality - just makes it a little more readable.

Reply:
Done